### ISSUE TITLE
[FEATURE] Update Implicit Function APIs to Accept and/or Mutate a DataFrame

### ISSUE THREAD

**Issue by TheCedarPrince** [JuliaHealth/OMOPCDMCohortCreator.jl#53]
This has been a feature I have been thinking about for a while wanting. I think all species of functions within this package should be able to accept a DataFrame and, depending on the function, know how to index that DataFrame to automatically retrieve information required. Additionally, functions maybe should automatically join results onto a passed in DataFrame. 

The reason for these changes is that I often want to use the pattern:

```julia
using Chain
using DataFrames
using OMOPCDMCohortCreator

@chain patient_df begin
  GetPatientGender
  GetPatientRace
  GetPatientAgeGroup
  _[:, Not(:person_id)]
  groupby(_, names(_))
end
```

or even

```julia
Characterize(x) = (GetPatientGender ∘ GetPatientRace ∘ GetPatientAgeGroup)(x)
```

To do very quick, rapid analyses and to re-use analyses over and over again clearly and explicitly. Not sure how much of the API should change as a result of this fix but would lend itself much better to composed functions and composition.

**Comment by TheCedarPrince** [comment]
Here is what we discussed in our call:

```julia

# Existing dispatch of working with person ids
GetPatientGender([1, 2, 3], conn)

# Issue idea
using DataFrames 
df = DataFrame([1, 2, 3], cols = [:person_id])

GetPatientGender(df, conn)

# Dispatch function "knows" what column it is expecting to see from the DataFrame
function GetPatientGender(df::DataFrame, conn; ...)
  ids = df.person_id
  conn = conn

  GetPatientGender(

end
```

Let me know if you have any questions -- thanks!

**Comment by TheCedarPrince** [comment]
Closed by #54
